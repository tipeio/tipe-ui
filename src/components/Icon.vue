<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
