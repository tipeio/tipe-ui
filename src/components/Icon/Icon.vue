<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
