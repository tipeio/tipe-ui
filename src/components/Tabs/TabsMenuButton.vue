<template>
  <button
    :class="rootClassObject"
    :data-tipe-ui="$options.name"
    @click="$emit('click')"
  >
    <div :class="layoutClassObject">
      <div
        v-if="icon"
        class="icon-container"
      >
        <tipe-icon :icon="icon"/>
      </div>
      <div class="label">
        {{ label }}
      </div>
    </div>
  </button>
</template>

<script>
import TabsMenuButtonShape from '@/types/TabsMenuButton'
import TipeIcon from '@/components/Icon'

export default {
  name: 'TipeTabsMenuButton',
  components: {
    TipeIcon
  },
  props: TabsMenuButtonShape,
  computed: {
    rootClassObject() {
      return {
        active: this.active
      }
    },
    layoutClassObject() {
      return {
        'icon-layout': this.icon,
        'label-layout': !this.icon
      }
    }
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeTabsMenuButton'] {
  height: 2.6875rem;
  margin: 0;
  padding: 0;
  border: none;
  outline: none;
  background-color: transparent;
  box-sizing: border-box;
  border-bottom: 0.1875rem solid transparent;
}

.label-layout {
  display: flex;
  align-items: center;
  padding-left: 0.8125rem;
  box-sizing: border-box;
}

.icon-layout {
  display: grid;
  grid-template-columns: 2rem auto;
  align-items: center;
}

.icon-container {
  display: flex;
  justify-content: center;
}

.label {
  display: flex;
  justify-content: flex-start;
  font-size: 0.8125rem;
  letter-spacing: 0.02rem;
  color: var(--text-gray);
  box-sizing: border-box;
  padding-right: 2.5rem;
}

.active {
  border-bottom: 0.1875rem solid var(--purple);
}
</style>
