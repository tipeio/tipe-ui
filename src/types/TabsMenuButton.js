import vueTypes from 'vue-types'

export default {
  icon: vueTypes.string.def(''),
  label: vueTypes.string.def(''),
  active: vueTypes.bool.def(false)
}
