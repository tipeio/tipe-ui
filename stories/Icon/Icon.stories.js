import { storiesOf } from '@storybook/vue'

import TipeIcon from '../../src/components/Icon'

const style = () => ({
  height: '1rem',
  width: '1rem'
})

const icons = [
  'check',
  'document',
  'folder',
  'key',
  'number',
  'share',
  'switch',
  'user',
  'color',
  'edit',
  'gear',
  'link',
  'pin',
  'sign-out',
  'text',
  'view',
  'copy',
  'flow',
  'grid',
  'list',
  'search',
  'star-active',
  'trash',
  'wrench',
  'credit-card',
  'folder-open',
  'group',
  'markdown',
  'media',
  'send',
  'star',
  'user-circle',
  'options',
  'plus',
  'cross',
  'arrow-down',
  'arrow-up',
  'arrow-right',
  'arrow-left'
]

storiesOf('Icon', module).add('icons', () => ({
  components: { TipeIcon },
  computed: { style },
  template:
    '<div>' +
    icons.reduce((template, icon) => {
      return (
        template +
        `<tipe-icon :style="style" icon="${icon}" color="rgb(255,0,0)" />`
      )
    }, '') +
    '</div>'
}))
