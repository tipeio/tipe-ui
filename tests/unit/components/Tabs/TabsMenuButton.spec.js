import { shallowMount } from '@vue/test-utils'
import TabsMenuButton from '@/components/Tabs/TabsMenuButton.vue'
import Icon from '@/components/Icon'
import faker from 'faker'

describe('TabsMenuButton', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const wrapper = shallowMount(TabsMenuButton)

      expect(wrapper).toMatchSnapshot()
    })

    it('<root> - has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(TabsMenuButton)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeTabsMenuButton')
    })

    it('<root> - should be button tag', () => {
      const wrapper = shallowMount(TabsMenuButton)

      expect(wrapper.is('button')).toBe(true)
    })
  })

  describe(':props', () => {
    it(':icon - should pass icon prop to the Icon component', () => {
      const propsData = { icon: faker.lorem.word() }
      const wrapper = shallowMount(TabsMenuButton, { propsData })

      expect(wrapper.find(Icon).props().icon).toEqual(propsData.icon)
    })

    it(':label - should be rendered', () => {
      const propsData = { label: faker.lorem.word() }
      const wrapper = shallowMount(TabsMenuButton, { propsData })

      expect(wrapper.text()).toBe(propsData.label)
    })

    it(':active(fasle) - should not render active class', () => {
      const propsData = { active: false }
      const wrapper = shallowMount(TabsMenuButton, { propsData })

      expect(wrapper.classes()).not.toContain('active')
    })

    it(':active(true) - should render active class', () => {
      const propsData = { active: true }
      const wrapper = shallowMount(TabsMenuButton, { propsData })

      expect(wrapper.classes()).toContain('active')
    })
  })

  describe('@events', () => {
    it('@click - should emit click event', () => {
      const wrapper = shallowMount(TabsMenuButton)

      wrapper.trigger('click')
      expect(wrapper.emitted().click).toBeTruthy()
    })
  })
})
