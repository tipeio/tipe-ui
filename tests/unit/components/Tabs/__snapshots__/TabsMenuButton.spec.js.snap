// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsMenuButton <template> matches previous snapshot 1`] = `
<button data-tipe-ui="TipeTabsMenuButton" class="">
  <div class="label-layout">
    <!---->
    <div class="label">

    </div>
  </div>
</button>
`;
